src/libsyntax/parse/parser.rs

```diff
@@ -5131,6 +5183,34 @@ impl<'a> Parser<'a> {
                     span: lo.to(self.prev_span),
                 });
                 seen_binding = true;
+            } else if self.check_const_arg() {
+                // TODO(const_generics): Numbers and expressions in braces are
+                // easy to check for, but bare Idents for `const`s or const
+                // params are indistinguishable from Idents for type params.
+                // There are two possibilities I can see to resolving the
+                // parsing problem. Either mark all const args with const (or
+                // something else, it doesn't really matter) or merge const args
+                // and type args together at the AST level and separate them
+                // later on, probably when lowering to the HIR.
```

Comment:
If `S<MY_CONST>` is supposed to work, then it can use same approach as identifiers in patterns.
In AST it's `PatKind::Ident`, but in HIR it's turned into a proper pattern based on resolution, for const arguments it would be something like `ast::GenericArg::Ident`.
I agree this is not necessary to do in this PR though.